### ISSUE TITLE
Make it work for RTSP H264

### ISSUE THREAD

**Issue by liutingdu** [evercam/evercam-capture-android#4]

**Comment by GNUDimarik** [comment]
@liutingdu Please provide description

**Comment by liutingdu** [comment]
@GNUDimarik 

Please see the issue description that I just updated, we discussed about this two weeks ago by Email, and I've included those details as description here.

Some more details: The Android API Camera.PreviewCallback() is able to send camera frames in byte[] continuously, that's the API we use for JPG images, I think it can be used for RTSP as well. But seems that API is deprecated now, if we are going to work on this, we might need to check out the latest Android API, there should be a new one provided.

Let me know if there are any questions!

**Comment by GNUDimarik** [comment]
@liutingdu @marcoherbst  
We can do it via ffmpeg, or may be via open max, open max has hw acceleration under android.
If I get it right you have some server where you want to send frames, right?

**Comment by liutingdu** [comment]
No, the app should works as a RTSP server, and produce RTSP stream when requested.

We have a SSH server that only does port forwarding - for example, the port on the phone is 192.168.1.10:80, and the SSH server forward it to ssh.evercam.io:10080, so that it's accessible on the Internet, not only locally. But it's nothing to do with this task, this task is only about expose an RTSP URL on the Android phone that serves the camera stream

**Comment by marcoherbst** [comment]
@mosic, I'd like to include you here because:

Naturally, @liuting is trying to solve the problem to fit into today's
Evercam which ingests RTSP but, would this not be the time to add adaptive
streaming / HLS / Mpeg-Dash ?

I'd like you to consider if we might be able to jump over h264 rtsp
straight to something better.

Also, @scandox you might have some input.

I suspect the solution here is to use someone else's ready made solution,
not to build our own.

Google github /stackoverflow is full of resources for this, it's a case of
picking one , I think.

M





On 7 July 2015 at 15:10, Liuting Du <notifications@github.com> wrote:

> No, the app should works as a RTSP server, and produce RTSP stream when
> requested.
>
> We have a SSH server that only does port forwarding - for example, the
> port on the phone is 192.168.1.10:80, and the SSH server forward it to
> ssh.evercam.io:10080, so that it's accessible on the Internet, not only
> locally. But it's nothing to do with this task, this task is only about
> expose an RTSP URL on the Android phone that serves the camera stream
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/evercam/evercam-capture-android/issues/4#issuecomment-119215717>
> .
>



-- 
Marco Herbst
*Evercam.io <http://www.evercam.io/>* - Cameras Unleashed
Mobile: +353 86 196 0451
Skype: marco.a.herbst

**Comment by marcoherbst** [comment]
2 links I came across:
https://www.google.ie/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=gstreamer%20to%20stream%20my%20camera%20on%20android

http://www.androidhive.info/2014/06/android-streaming-live-camera-video-to-web-page/

M

**Comment by liutingdu** [comment]
The tutorial you mentioned is using the libstreaming library: https://github.com/fyhertz/libstreaming 

Seems it's a quite popular open source library and it's the same library that [Spydroid](https://github.com/fyhertz/spydroid-ipcamera) is using. They are from the same author.

**Comment by GNUDimarik** [comment]
I know about  Spydroid, it's very slowly

**Comment by liutingdu** [comment]
@GNUDimarik  It's great that you know it! That's what we want to implement for Evercam Capture app - stream video from Android, have you got any solutions better than this libstreaming library? I tried the Spydroid app and seems it's not working with my nexus 5.. Thanks!

**Comment by GNUDimarik** [comment]
@liutingdu  Do you mean new code?

**Comment by liutingdu** [comment]
@GNUDimarik Here we are talking about a different app - Evercam Capture. What's the 'new code' you are referring to?

**Comment by GNUDimarik** [comment]
@liutingdu Sorry, I wanted to reply in other bug :-)
